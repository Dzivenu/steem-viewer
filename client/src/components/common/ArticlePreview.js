import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import _ from 'lodash';
import EN from '../../constants/translations/en';
import Link from './Link';
import { COLORS } from '../../constants/styles';
import Article from '../article/Article';

const Container = styled.div`
  border: 1px solid ${COLORS.WHITE.WHITE_SMOKE};
  box-shadow: 0 2px 4px 0 rgba(0,0,0,.05);
  margin: 1em;
  width: 600px;
`;

const ImageContainer = styled.div`
  display: flex;
  justify-content: center;
`;
const PreviewImage = styled.img`
  width: auto;
  max-height: 475px;
  max-width: 100%;
`;

const Header = styled.div`
  border-bottom: 1px solid ${COLORS.WHITE.WHITE_SMOKE};
  padding: 1em;
`;

const Content = styled.div`
`;

const Title = styled.a`
  cursor: pointer;
  display: block;
  color: ${COLORS.BLUE.EAST_BAY};
  font-weight: 400;
  font-size: 1.5em;
  padding: 1em .5em;
`;

const Footer = styled.div`
  border-top: 1px solid ${COLORS.WHITE.WHITE_SMOKE};
  display: flex;
`;

const FooterContent = styled.div`
  padding: 1em;
  border-right: 1px solid ${COLORS.WHITE.WHITE_SMOKE};
  
  &:last-of-type {
    border-right: 0;
  }
`;

const Icon = styled.i`
  color: ${COLORS.BLUE.LINK_WATER};
  margin-right: .25em;
`;

class ArticlePreview extends Component {
  static propTypes = {
    author: PropTypes.string,
    tags: PropTypes.array,
    images: PropTypes.array,
    category: PropTypes.string,
    title: PropTypes.string,
    body: PropTypes.string,
    likes: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    payout: PropTypes.string,
    comments: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    permlink: PropTypes.string,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  };

  constructor(props) {
    super(props);

    this.state = {
      displayArticle: false,
    };

    this.showArticle = this.showArticle.bind(this);
    this.hideArticle = this.hideArticle.bind(this);
  }

  showArticle() {
    this.setState({ displayArticle: true });
    document.body.style = 'overflow:hidden;';
  }

  hideArticle() {
    this.setState({ displayArticle: false });
    document.body.style = 'overflow:initial;';
  }

  render() {
    const {
      images,
      author,
      category,
      title,
      likes,
      payout,
      comments,
      permlink,
      body,
      tags,
      id,
    } = this.props;
    const { displayArticle } = this.state;
    const firstImage = _.head(images);

    return (
      <Container>
        <Header>
          <Link>{author}</Link>
          {` ${EN.in} `}
          <Link>{`#${category}`}</Link>
        </Header>
        <Content>
          <Title onClick={this.showArticle}>{title}</Title>
        </Content>
        <ImageContainer>
          <Link onClick={this.showArticle}>
            <PreviewImage src={firstImage} />
          </Link>
        </ImageContainer>
        <Footer>
          <FooterContent>
            <Icon className={'fa fa-thumbs-up'} />{`${likes} ${EN.likes}`}
          </FooterContent>
          <FooterContent><Icon className={'fa fa-usd'} />{payout}</FooterContent>
          <FooterContent><Icon className={'fa fa-comments'} />{comments}</FooterContent>
        </Footer>
        {displayArticle &&
          <Article
            hideArticle={this.hideArticle}
            author={author}
            category={category}
            permlink={permlink}
            title={title}
            body={body}
            tags={tags}
            id={id}
          />}
      </Container>
    );
  }
}

export default ArticlePreview;
