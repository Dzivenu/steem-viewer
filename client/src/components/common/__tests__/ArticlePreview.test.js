import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArticlePreview from '../ArticlePreview';

Enzyme.configure({ adapter: new Adapter() });

describe('(Component) ArticlePreview', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        author: 'test',
        tags: [],
        images: [],
        category: 'test category',
        title: 'test title',
        body: 'test body',
        likes: 0,
        payout: '0',
        comments: 0,
        permlink: 'test',
        id: 0,
      };
      const wrapper = shallow(<ArticlePreview {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
