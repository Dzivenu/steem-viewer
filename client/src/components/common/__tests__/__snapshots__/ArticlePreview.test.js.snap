// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ArticlePreview with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlePreview
    author="test"
    body="test body"
    category="test category"
    comments={0}
    id={0}
    images={Array []}
    likes={0}
    payout="0"
    permlink="test"
    tags={Array []}
    title="test title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.a>
                    test
          </styled.a>
           in 
          <styled.a>
                    #test category
          </styled.a>
</styled.div>,
        <styled.div>
          <styled.a
                    onClick={[Function]}
          >
                    test title
          </styled.a>
</styled.div>,
        <styled.div>
          <styled.a
                    onClick={[Function]}
          >
                    <styled.img
                              src={undefined}
                    />
          </styled.a>
</styled.div>,
        <styled.div>
          <styled.div>
                    <styled.i
                              className="fa fa-thumbs-up"
                    />
                    0 Likes
          </styled.div>
          <styled.div>
                    <styled.i
                              className="fa fa-usd"
                    />
                    0
          </styled.div>
          <styled.div>
                    <styled.i
                              className="fa fa-comments"
                    />
                    0
          </styled.div>
</styled.div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.a>
              test
</styled.a>,
            " in ",
            <styled.a>
              #test category
</styled.a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "test",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          " in ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "#test category",
            },
            "ref": null,
            "rendered": "#test category",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.a
            onClick={[Function]}
>
            test title
</styled.a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "test title",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "test title",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.a
            onClick={[Function]}
>
            <styled.img
                        src={undefined}
            />
</styled.a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.img
              src={undefined}
/>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <styled.i
                            className="fa fa-thumbs-up"
              />
              0 Likes
</styled.div>,
            <styled.div>
              <styled.i
                            className="fa fa-usd"
              />
              0
</styled.div>,
            <styled.div>
              <styled.i
                            className="fa fa-comments"
              />
              0
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.i
                  className="fa fa-thumbs-up"
/>,
                "0 Likes",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "className": "fa fa-thumbs-up",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "0 Likes",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.i
                  className="fa fa-usd"
/>,
                "0",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "className": "fa fa-usd",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "0",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.i
                  className="fa fa-comments"
/>,
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "className": "fa fa-comments",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              0,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.a>
                        test
            </styled.a>
             in 
            <styled.a>
                        #test category
            </styled.a>
</styled.div>,
          <styled.div>
            <styled.a
                        onClick={[Function]}
            >
                        test title
            </styled.a>
</styled.div>,
          <styled.div>
            <styled.a
                        onClick={[Function]}
            >
                        <styled.img
                                    src={undefined}
                        />
            </styled.a>
</styled.div>,
          <styled.div>
            <styled.div>
                        <styled.i
                                    className="fa fa-thumbs-up"
                        />
                        0 Likes
            </styled.div>
            <styled.div>
                        <styled.i
                                    className="fa fa-usd"
                        />
                        0
            </styled.div>
            <styled.div>
                        <styled.i
                                    className="fa fa-comments"
                        />
                        0
            </styled.div>
</styled.div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.a>
                test
</styled.a>,
              " in ",
              <styled.a>
                #test category
</styled.a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": [Function],
            },
            " in ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "#test category",
              },
              "ref": null,
              "rendered": "#test category",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.a
              onClick={[Function]}
>
              test title
</styled.a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "test title",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "test title",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.a
              onClick={[Function]}
>
              <styled.img
                            src={undefined}
              />
</styled.a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <styled.img
                src={undefined}
/>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.i
                                className="fa fa-thumbs-up"
                />
                0 Likes
</styled.div>,
              <styled.div>
                <styled.i
                                className="fa fa-usd"
                />
                0
</styled.div>,
              <styled.div>
                <styled.i
                                className="fa fa-comments"
                />
                0
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.i
                    className="fa fa-thumbs-up"
/>,
                  "0 Likes",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "className": "fa fa-thumbs-up",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "0 Likes",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.i
                    className="fa fa-usd"
/>,
                  "0",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "className": "fa fa-usd",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "0",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.i
                    className="fa fa-comments"
/>,
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "className": "fa fa-comments",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                0,
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
